File: Capítulo_7/ECMAScript 2015 – a classe Set/Operações com a classe Set da ES2015/Simulando a operação de intersecção/intersection.js
// Usaremos os dois conjuntos a seguir em nossos exemplos:
const setA = new Set();
setA.add(1);
setA.add(2);
setA.add(3);
const setB = new Set();
setB.add(2);
setB.add(3);
setB.add(4);

const intersection = (set1, set2) => {
    const intersectionSet = new Set();
    set1.forEach(value => {
        if (set2.has(value)) {
            intersectionSet.add(value);
        }
    });
    return intersectionSet;
};
console.log(intersection(setA, setB)); // Set(2) { 2, 3 }

File: Capítulo_7/ECMAScript 2015 – a classe Set/Operações Set ES2015/Simulando a operação de intersecção/Código em Python/intersection.py
# Usaremos os dois conjuntos a seguir em nossos exemplos:
setA = set()
setA.add(1)
setA.add(2)
setA.add(3)
setB = set()
setB.add(2)
setB.add(3)
setB.add(4)

def intersection(set1, set2):
    intersection_set = set()
    for value in set1:
        if value in set2:
            intersection_set.add(value)
    return intersection_set

intersection = intersection(setA, setB)

print(intersection) # { 2, 3 }
